File: AkkaSupervisorExample/src/main/java/org/akka/essentials/supervisor/example3/MyActorSystem.java
package org.akka.essentials.supervisor.example3;

import akka.actor.ActorRef;
import akka.actor.ActorSystem;
import akka.actor.Props;

public class MyActorSystem {

	/**
	 * @param args
	 * @throws Exception
	 */
	public static void main(String[] args) throws Exception {
		ActorSystem system = ActorSystem.create("faultTolerance");

		ActorRef supervisor = system.actorOf(new Props(SupervisorActor.class),
				"supervisor");

		supervisor.tell(Integer.valueOf(10));
		supervisor.tell("10");

		Thread.sleep(5000);

		supervisor.tell(Integer.valueOf(10));

		system.shutdown();
	}

}

File: AkkaSupervisorExample/src/main/scala/org/akka/essentials/supervisor/example3/MyActorSystem.scala
package org.akka.essentials.supervisor.example3
import akka.actor.ActorSystem
import akka.actor.Props
import akka.pattern.ask
import akka.util.duration._
import akka.dispatch.Await
import akka.util.Timeout
import akka.actor.ActorLogging
import akka.actor.Actor
import akka.actor.OneForOneStrategy
import akka.dispatch.Future
import akka.actor.SupervisorStrategy._
import akka.actor.ActorRef

case class Result
case class DeadWorker
case class RegisterWorker(val worker: ActorRef, val supervisor: ActorRef)

object MyActorSystem extends App {

  val system = ActorSystem("faultTolerance")
  val log = system.log

  val supervisor = system.actorOf(Props[SupervisorActor], name = "supervisor")

  var mesg: Int = 8;
  supervisor ! mesg

  supervisor ! "Do Something"

  Thread.sleep(4000)

  supervisor ! mesg

  system.shutdown();

}